tests/subsys/logging/log_core_additional/prj.conf

```diff
@@ -9,8 +9,9 @@ CONFIG_LOG_BUFFER_SIZE=1024
 CONFIG_CBPRINTF_FP_SUPPORT=y
 CONFIG_CBPRINTF_LIBC_SUBSTS=y
 CONFIG_ZTEST_STACK_SIZE=4096
-CONFIG_LOG_PROCESS_THREAD=n
+CONFIG_LOG_MODE_DEFERRED=y
```

Comment:
Test was written in a way that it expects no thread and local `log_test_process()` is called in the tests so now it is not tested anymore. You can add it as second configuration in `testcase.yaml`.

Reply:
I was under the impression that logging.sync tested this case (+CONFIG_LOG_MODE_IMMEDIATE is enabled), without my change the differences in the config file between sync and async is very minimal, and code written for deferred case (w/ thread) is not executed at all, hence why tests didn't fail with deprecated APIs:

```
--- logging.async/zephyr/.config	2024-04-17 13:47:21.918718205 -0700
+++ logging.sync/zephyr/.config	2024-04-17 13:47:21.921718222 -0700
@@ -22,7 +22,7 @@
 # CONFIG_COUNTER is not set
 # CONFIG_PM_DEVICE is not set
 CONFIG_TICKLESS_KERNEL=y
-CONFIG_ZTEST_STACK_SIZE=4096
+CONFIG_ZTEST_STACK_SIZE=1024
 # CONFIG_FPU is not set
 # CONFIG_ZTEST_NO_YIELD is not set
 # CONFIG_GPIO is not set
@@ -795,11 +795,11 @@
 # CONFIG_CBPRINTF_NANO is not set
 CONFIG_CBPRINTF_FULL_INTEGRAL=y
 # CONFIG_CBPRINTF_REDUCED_INTEGRAL is not set
-CONFIG_CBPRINTF_FP_SUPPORT=y
+# CONFIG_CBPRINTF_FP_SUPPORT is not set
 # CONFIG_CBPRINTF_FP_A_SUPPORT is not set
 # CONFIG_CBPRINTF_FP_ALWAYS_A is not set
 CONFIG_CBPRINTF_N_SPECIFIER=y
-CONFIG_CBPRINTF_LIBC_SUBSTS=y
+# CONFIG_CBPRINTF_LIBC_SUBSTS is not set
 # CONFIG_CBPRINTF_PACKAGE_LOG_LEVEL_OFF is not set
 # CONFIG_CBPRINTF_PACKAGE_LOG_LEVEL_ERR is not set
 # CONFIG_CBPRINTF_PACKAGE_LOG_LEVEL_WRN is not set
@@ -808,7 +808,6 @@
 CONFIG_CBPRINTF_PACKAGE_LOG_LEVEL_DEFAULT=y
 CONFIG_CBPRINTF_PACKAGE_LOG_LEVEL=3
 # CONFIG_CBPRINTF_PACKAGE_LONGDOUBLE is not set
-# CONFIG_CBPRINTF_STATIC_PACKAGE_CHECK_ALIGNMENT is not set
 CONFIG_CBPRINTF_CONVERT_CHECK_PTR=y
 # end of OS Support Library
 
@@ -979,7 +978,7 @@
 #
 # Processing
 #
-# CONFIG_LOG_PRINTK is not set
+CONFIG_LOG_PRINTK=y
 CONFIG_LOG_TRACE_SHORT_TIMESTAMP=y
 # CONFIG_LOG_TIMESTAMP_64BIT is not set
 # end of Processing
```

Reply:
Immediate mode is different - each log is processed in the context of the call and deferred has two flavors: much more common where logging has own thread in which logs are processed and mode where this thread is not created and it's up to the user to call processing function, e.g. it can be used in no multithreading mode where in the main loop you call `log_process()`. `CONFIG_LOG_PROCESS_THREAD=n` tests this setup.

Reply:
Got it, added new test configuration for the ```CONFIG_LOG_PROCESS_THREAD=y``` cases.